extensions/amp-sidebar/1.0/toolbar.js

```diff
@@ -88,6 +94,15 @@ export class Toolbar {
     } else {
       this.hideToolbar_();
     }
+
+    // Use the placeholder to fill the height of the toolbar
+    this.vsync_.mutate(() => {
+      if (this.placeholder_) {
```

Comment:
This should be done in attempt show , you dont want to display the compensatingElement when toolbar is hidden. 

1. Check if there is a compensation element , if not create one and assign the height
2. else just assign the height
3. When you close the toolbar this has to be hidden